app/lib/frontend/templates.dart

```diff
@@ -768,6 +735,45 @@ class TemplateService {
     });
     return parsedTemplate.renderString(values);
   }
+
+  String renderPlatformTabs({
+    String platform,
+    SearchQuery searchQuery,
+  }) {
+    final String currentPlatform =
+        platform ?? searchQuery?.platformPredicate?.single;
+    Map platformTabData(String tabText, String tabPlatform) {
+      final Map<String, String> params = <String, String>{};
+      if (tabPlatform != null) {
+        params['platform'] = tabPlatform;
+      }
+      String url;
+      if (searchQuery != null) {
+        final newQuery = searchQuery.change(
+            platformPredicate: tabPlatform == null
+                ? new PlatformPredicate()
+                : new PlatformPredicate(required: [tabPlatform]));
+        url = newQuery.toSearchLink();
+      } else {
+        url = new Uri(path: '/packages', queryParameters: params).toString();
```

Comment:
Move the computation of `params` into this `} else {` branch, since it's only used here.